client/python/gradio_client/client.py

```diff
@@ -451,17 +463,53 @@ def _use_websocket(self, dependency: Dict) -> bool:
 
     async def _ws_fn(self, data, hash_data, helper: Communicator):
         async with websockets.connect(  # type: ignore
-            self.ws_url, open_timeout=10, extra_headers=self.headers
+            self.ws_url,
+            open_timeout=10,
+            extra_headers=self.headers,
+            max_size=1024 * 1024 * 1024,
         ) as websocket:
             return await utils.get_pred_from_ws(websocket, data, hash_data, helper)
 
 
 class Job(Future):
     """A Job is a thin wrapper over the Future class that can be cancelled."""
 
-    def __init__(self, future: Future, communicator: Communicator | None = None):
+    def __init__(
+        self,
+        future: Future,
+        communicator: Communicator | None = None,
+        is_generator: bool = False,
+    ):
         self.future = future
         self.communicator = communicator
+        self.is_generator = is_generator
+
+    def outputs(
+        self,
+    ):
+        if not self.communicator:
+            return []
+        else:
+            with self.communicator.lock:
+                return self.communicator.job.outputs
+
+    def result(self, timeout: float | None = None):
+        if self.is_generator:
+            # Coerce to inf here because the parent class can't handle inf
+            timeout = timeout or math.inf
```

Comment:
Tiniest of nits: using `float("inf")` will allow you to not have to `import math`